components/form/TextField.tsx

```diff
@@ -36,26 +36,69 @@ export const TextField = React.forwardRef<RNView, TextFieldProps>(({name, label,
     [textTransform, field],
   );
 
+  const onBlur = useCallback(() => {
+    setFocused(false);
+    field.onBlur();
+  }, [field]);
+
+  const onFocus = useCallback(() => {
+    setFocused(true);
+  }, []);
+
+  const textFieldRef = useRef<TextInput>(null);
+  const [isFocused, setFocused] = useState(() => textFieldRef?.current?.isFocused ?? false);
+
+  useLayoutEffect(() => {
+    setFocused(textFieldRef?.current?.isFocused ?? false);
+  }, []);
```

Comment:
Does this dependency array need `textFieldRef`?

Reply:
No, refs are not to be used as dependencies. Neither is the setter of `useState`. I'll try to dig up the docs, but that's why there's no lint warning here or on line 57.

Reply:
So I can't find any written docs on this (I know I've read some), but here's the unit tests for the lint rule describing how the `setState` and `useRef` values are considered to be static.

https://github.com/facebook/react/blob/d3037405c4c9ce8158ef9017e960ec97fb089df4/packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js#L644-L656

Reply:
I see, interesting - so they're guaranteed static, so it's not necessary to put them in. Neat, TIL.